roles/confluent.ssl/tasks/custom_certs.yml

```diff
@@ -27,15 +42,20 @@
     keytool -noprompt -keystore {{truststore_path}} \
       -storetype pkcs12 \
       -alias CARoot \
-      -import -file /var/ssl/private/generation/{{ssl_ca_cert_filepath|basename}} \
+      -import -file {{ssl_ca_cert}} \
       -storepass {{truststore_storepass}} \
       -keypass {{truststore_storepass}} {{extra_args}}
+  when: not ssl_add_ca_chain
+
+- name: Create Truststore and Import the CA Chain
+  include_tasks: import_ca_chain.yml
+  when: ssl_add_ca_chain
 
 - name: Put Key and Signed Cert into pkcs12 Format
   shell: |
     openssl pkcs12 -export \
-      -in /var/ssl/private/generation/signed.crt \
-      -inkey /var/ssl/private/generation/key.pem \
+      -in {{ssl_signed_cert}} \
+      -inkey {{ssl_key}} \
       -passin pass:{{ssl_key_password}} \
```

Comment:
didnt you have a requirement that this password could be optionally skipped?
I think it can be handles with a 'when: ssl_key_password is (not) defined`

Reply:
Yeah. I’ll add that 👍